common/tier4_autoware_utils/include/tier4_autoware_utils/math/precompute.hpp

```diff
@@ -0,0 +1,61 @@
+// Copyright 2020 Tier IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef TIER4_AUTOWARE_UTILS__MATH__PRECOMPUTE_HPP_
+#define TIER4_AUTOWARE_UTILS__MATH__PRECOMPUTE_HPP_
+
+#include <cmath>
+#include <vector>
+
+namespace tier4_autoware_utils
+{
+
+class PrecomputedTrigFunc
+{
+  const float PI = std::acos(-1);
```

Comment:
Is there any reason not to use `tier4_autoware_utils::pi` or `M_PI`?
https://github.com/autowarefoundation/autoware.universe/blob/b28d792dee2379b7e9561ec256dc0052c60d127b/common/tier4_autoware_utils/include/tier4_autoware_utils/math/constants.hpp#L20